src/promptflow/promptflow/executor/_service/apis/execution.py

```diff
@@ -5,52 +5,95 @@
 from promptflow.executor._service.contracts.execution_request import FlowExecutionRequest, NodeExecutionRequest
-from promptflow.executor._service.utils.service_utils import get_log_context, set_environment_variables
+from promptflow.executor._service.utils.process_utils import invoke_function_in_process
+from promptflow.executor._service.utils.service_utils import (
+    get_executor_version,
+    get_service_log_context,
+    set_environment_variables,
+    update_operation_context,
+)
 from promptflow.executor.flow_executor import FlowExecutor, execute_flow
 from promptflow.storage._run_storage import DefaultRunStorage
 
-router = APIRouter()
+router = APIRouter(prefix="/execution")
 
 
-@router.post("/execution/flow")
+@router.post("/flow")
 async def flow_execution(request: Request, flow_request: FlowExecutionRequest):
-    OperationContext.get_instance().update(dict(request.headers))
+    update_operation_context(dict(request.headers))
+    request_id = OperationContext.get_instance().request_id
+    executor_version = get_executor_version()
+    with get_service_log_context(flow_request):
+        service_logger.info(
+            f"Received flow execution request, flow run id: {flow_request.run_id}, "
+            f"request id: {request_id}, executor version: {executor_version}."
+        )
+        try:
+            result = await invoke_function_in_process(flow_request, request.headers, flow_test)
+            service_logger.info(f"Completed flow execution request, flow run id: {flow_request.run_id}.")
+            return result
+        except Exception as ex:
+            error_type_and_message = (f"({ex.__class__.__name__}) {ex}",)
+            service_logger.error(
+                f"Failed to execute flow, flow run id: {flow_request.run_id}. Error: {error_type_and_message}"
+            )
+            raise ex
+
+
+@router.post("/node")
+async def node_execution(request: Request, node_request: NodeExecutionRequest):
+    update_operation_context(dict(request.headers))
+    request_id = OperationContext.get_instance().request_id
+    executor_version = get_executor_version()
+    with get_service_log_context(node_request):
+        service_logger.info(
+            f"Received node execution request, node name: {node_request.node_name}, "
+            f"request id: {request_id}, executor version: {executor_version}."
+        )
+        try:
+            result = await invoke_function_in_process(node_request, request.headers, single_node_run)
+            service_logger.info(f"Completed node execution request, node name: {node_request.node_name}.")
+            return result
+        except Exception as ex:
+            error_type_and_message = (f"({ex.__class__.__name__}) {ex}",)
+            service_logger.error(
+                f"Failed to execute node, node name: {node_request.node_name}. Error: {error_type_and_message}"
+            )
+            raise ex
+
+
+async def flow_test(flow_request: FlowExecutionRequest):
```

Comment:
Do we need to change working directory for flow execution?

Reply:
![image](https://github.com/microsoft/promptflow/assets/111329184/888b9f49-c803-43d9-be34-928641cf5bd4)
We do this in `execute_flow`.